.travis.yml

```diff
@@ -44,30 +44,28 @@ matrix:
      language: python
      python: 3.5
      env: TESTENV=build_console
-     before_install: make deps
+     before_install:
+       - make deps
+       - pip install -r requirements_dev.txt
      script: make build_console
 
    - os: osx
      osx_image: Xcode 8.3.3
-     env: TESTENV=mac TOXENV=py27,py37
      language: generic
+     env: TESTENV=mac TOXENV=py27,py35,py37,gui35
      before_install:
-        - export PATH="/usr/local/opt/python@2/libexec/bin:$PATH"
+        - export PATH="/Users/travis/miniconda3/bin:$PATH"
         - make deps
-     before_script:
-        - export PATH="/usr/local/bin/:./:$PATH"
-        - export DISPLAY=:0.0
-        - Xvfb ${DISPLAY} -screen 0 800x600x16 & PID=$!
-        - sleep 3  # give xvfb some time to start
      script:
-        - tox -v
```

Comment:
This is a heavily subjective opinion, feel free to ignore, but wouldn't it be a bit simpler to do just the conda setup in the Makefile, and call that target from here, and then tox directly from here? Now there are several targets in Makefile just for testing...

Reply:
I feel like the relationship between `.travis.yml`, the `Makefile`, and `tasks/setups.sh` are a bit ill-defined / muddy.  In general I'd to most data not in Travis since it makes testing much easier and makes the CI more portable to other systems (if we ever need to migrate).  I tested many of these things on a Mac OS X 10.13 VM.

Reply:
In any rate, `setup.sh` is definitely the "do bad stuff to your system" script, so installing conda there definitely felt like the right place.

Reply:
I agree that conda setup belongs to setup.sh. But I kind of liked the direct invocation of tox from travis. Just a personal opinion though, not necessarily the best way.